apps/rush-lib/src/cli/RushPnpmCommandLine.ts

```diff
@@ -0,0 +1,259 @@
+        const prefix: string = 'ERROR: ';
+        console.error('\n' + colors.red(PrintUtilities.wrapWords(prefix + error.message)));
+      }
+    }
+  }
+
+  private static _validatePnpmUsage(pnpmArgs: string[]): void {
+    if (pnpmArgs[0] === RUSH_SKIP_CHECKS_PARAMETER) {
+      pnpmArgs.shift();
+      // Ignore other checks
+      return;
+    }
+
+    if (pnpmArgs.length === 0) {
+      return;
+    }
+    const firstArg: string = pnpmArgs[0];
+
+    // Detect common safe invocations
+    if (pnpmArgs.indexOf('-h') >= 0 || pnpmArgs.indexOf('--help') >= 0 || pnpmArgs.indexOf('-?') >= 0) {
+      return;
+    }
+
+    if (pnpmArgs.length === 1) {
+      if (firstArg === '-v' || firstArg === '--version') {
+        return;
+      }
+    }
+
+    const BYPASS_NOTICE: string = `To bypass this check, add "${RUSH_SKIP_CHECKS_PARAMETER}" as the very first command line option.`;
+
+    if (!/^[a-z]+([a-z0-9\-])*$/.test(firstArg)) {
+      // We can't parse this CLI syntax
+      console.error(
+        colors.red(`Warning: The "rush-pnpm" wrapper expects a command verb before "${firstArg}"`) + '\n'
+      );
+      console.error(colors.cyan(BYPASS_NOTICE));
+      throw new AlreadyReportedError();
+    } else {
+      const commandName: string = firstArg;
+
+      // Also accept SKIP_RUSH_CHECKS_PARAMETER immediately after the command verb
+      if (pnpmArgs[1] === RUSH_SKIP_CHECKS_PARAMETER) {
+        pnpmArgs.splice(1, 1);
+        return;
+      }
+
+      if (pnpmArgs.indexOf(RUSH_SKIP_CHECKS_PARAMETER) >= 0) {
+        // We do not attempt to parse PNPM's complete CLI syntax, so we cannot be sure how to interpret
+        // strings that appear outside of the specific patterns that this parser recognizes
+        console.error(
+          colors.red(
+            PrintUtilities.wrapWords(
+              `Error: The "${RUSH_SKIP_CHECKS_PARAMETER}" option must be the first parameter for the "rush-pnpm" command.`
+            )
+          )
+        );
+        throw new AlreadyReportedError();
+      }
+
+      // Warn about commands known not to work
+      switch (commandName) {
+        case 'install':
+        case 'i':
+        case 'install-test':
+        case 'it':
+          console.error(
+            colors.red(
+              PrintUtilities.wrapWords(
+                `Error: The "pnpm ${commandName}" command is incompatible with Rush's environment.` +
+                  ` Use the "rush install" or "rush update" commands instead.`
+              )
+            ) + '\n'
+          );
+          console.error(colors.cyan(BYPASS_NOTICE));
+          throw new AlreadyReportedError();
+      }
+
+      const commandKind: CommandKind | undefined = RushPnpmCommandLine._knownCommandMap.get(commandName);
+      switch (commandKind) {
+        case CommandKind.KnownSafe:
+          break;
+        case CommandKind.ShowWarning:
+          console.log(
+            colors.yellow(
+              PrintUtilities.wrapWords(
```

Comment:
We could intentionally break the workplace state by deleting the `install.flag` file. This would leave the state as-is for the wrapper to operate, but invalidate the Rush commands that may rely strongly on the workspace install state.

Could also do this as a separate `CommandKind` since it's somewhere between `ShowWarning` and `Blocked` in terms of accessibility. Could also prevent clearing of state by passing a special flag. Just ideas.